docs/IntegrationWithExistingApps.md

```diff
@@ -720,6 +720,8 @@ public boolean onKeyUp(int keyCode, KeyEvent event) {
 
 That's it, your activity is ready to run some JavaScript code.
 
+> If your app is targeting the Android `api level 23` or greater, make sure you have, for the development build, the `overlay permission` enabled. You can check it with `Settings.canDrawOverlays(this);`. This is required because, if your app produces an error in the react native component, the error view is displayed above all the other windows. Due to the new permissions system, introduced in the api level 23, the user needs to approve it.
```

Comment:
@varungupta85 

> Is this required only to be able to show the dev menu and the red box

Yes. This is only required in the dev mode for the redbox and dev menu. So it'll work fine without the permission in production.

> Can I remove the SYSTEM_ALERT_WINDOW permission from the AndroidManifest.xml file and just request it programmatically?

Unlike other permissions, we cannot request this permission programmatically. In the code you linked to, I'm just directing the user to the settings screen where he can grant the permission.

> I am not sure why is the SYSTEM_ALERT_WINDOW permission added in the default AndroidManifest.xml file.

I'm not sure if you'll be able to grant the permission if it's not in the manifest. Probably it's added by the RN lib even if you remove it from your manifest. Maybe it's there in the manifest because of Buck. Not sure about this though.

> This is a bit unrelated and more for my own understanding. In this comment, you have mentioned to disable dev menu in production which is done in the default template. Where is this template located?

Earlier we had more stuff in `MainActivity`. Lots of refactor has been done, so it's not relevant anymore.